spec/database_spec.cr

```diff
@@ -114,6 +114,28 @@ describe DB::Database do
     end
   end
 
+  it "should checkout different connections until they are released" do
+    DummyDriver::DummyConnection.clear_connections
+    DB.open "dummy://localhost:1027?initial_pool_size=1&max_pool_size=2&retry_attempts=0" do |db|
+      the_first_cnn = uninitialized DB::Connection
+      the_second_cnn = uninitialized DB::Connection
+
+      the_first_cnn = db.checkout
+      the_second_cnn = db.checkout
+      the_second_cnn.should_not eq(the_first_cnn)
+      db.pool.is_available?(the_first_cnn).should be_false
+      db.pool.is_available?(the_second_cnn).should be_false
+
+      the_first_cnn.release
+      db.pool.is_available?(the_first_cnn).should be_true
+      db.pool.is_available?(the_second_cnn).should be_false
+
+      db.checkout.should eq(the_first_cnn)
+      the_first_cnn.release
+      the_second_cnn.release
+    end
+  end
```

Comment:
I'd add a spec that checks that if one of the checked-out connections executes a statement and closes the underlying resultset, then the connection is **not** returned to the pool.